### PR TITLE
Demo simu

### PR DESCRIPTION
**Pull request by rjacolin**
new features and bug fix about simulation

### CODE REVIEW

**Comment by rjacolin** [comment]
Sebz, Can you review this first list of features & bug fixes? Tell me how we sync about DM data generators integration.

File: lib/generators/index.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ module.exports = function(generatorConfig) {
 
         default:
             // Look for a custom generator
-            var CustomGenerator = require("custom-generators/" + generatorConfig.generator);
+            var CustomGenerator = require("../../custom-generators/" + generatorConfig.generator + ".js");
```
Comment (sebz) [inline lib/generators/index.js]: Correct for the `../../`, on the other hand, `+ ".js"` is not required.
I'll change that.

File: lib/engines/backToTheFuture.js [hunk 2]
```diff
@@ -72,7 +72,7 @@ B2TFEngine.prototype.getDataGenerationConfig = function(dataId) {
 };
 
 B2TFEngine.prototype.getApplicationData = function() {
```
Comment (sebz) [inline lib/engines/backToTheFuture.js]: I'll rename the function to `getGenerationData`

File: lib/engines/backToTheFuture.js [hunk 3]
```diff
@@ -40,24 +40,24 @@ B2TFEngine.prototype.generateData = function(system) {
...
+        for (var dataDescriptor in self.getApplicationData()){
             console.log("generate value for data:", dataDescriptor);
             try {
-                data[timestamp][dataDescriptor.id] = self.generateValue(dataDescriptor);
+                data[timestamp][dataDescriptor] = self.generateValue(dataDescriptor);
             } catch (e) {
                 console.error("# Data gen error:", e);
             }
-        });
+        };
```
Comment (sebz) [inline lib/engines/backToTheFuture.js]: No `;` here.
I'll change that.

File: index.js [hunk 4]
```diff
@@ -6,6 +6,7 @@ var jsonfile = require("jsonfile");
 var setup = jsonfile.readFileSync("./setup.json");
 var serverConfig = jsonfile.readFileSync("./config/airvantage/" + setup.dataCenter + ".json");
 var credentials = _.extend({}, setup.credentials, serverConfig.credentials);
+var companyUid = _.extend({}, setup.credentials, serverConfig.companyUid);
```
Comment (sebz) [inline index.js]: Not needed, the `companyUid` is already available through the `setup` variable.
I'll change that.

**Pull request merged**